src/core/utils/number/numberUtils.ts

```diff
@@ -156,7 +156,8 @@ function removeLeadingZeros(locale = navigator.language, value: string) {
     integerPart.length !== String(parseInt(integerPart)).length &&
     integerPart !== LOCALE_MINUS_SIGN &&
     integerPart !== LOCALE_NEGATIVE_ZERO &&
-    integerPart !== DEFAULT_NEGATIVE_ZERO
+    integerPart !== DEFAULT_NEGATIVE_ZERO &&
+    integerPart !== ""
```

Comment:
Should we do this instead ?

```
    integerPart &&
    integerPart.length !== String(parseInt(integerPart)).length &&
    integerPart !== LOCALE_MINUS_SIGN &&
    integerPart !== LOCALE_NEGATIVE_ZERO &&
    integerPart !== DEFAULT_NEGATIVE_ZERO
```

Reply:
I think we shouldn't. Because if you delete `integerPart` completely, `String(parseInt(integerPart))` will be `NaN` and they wouldn't be equal. So, this check wouldn't work.
```js
String(parseInt(integerPart)).length = 3
"NaN".length = 3
```

Reply:
I am not sure I understand. When we remove the integer part of the value, `integerPart` variable would be equal to empty string, `""`, is it not?

Reply:
Actually not. Let's think step by step.
If we delete `0` from `0.5`,
- `integerPart` = `""` (empty string)
- `parseInt("")` = `"NaN"`
- `String(NaN).length` = 3

![image](https://user-images.githubusercontent.com/44462260/123967327-df66c500-d9be-11eb-9f6c-6604b64514b2.png)
![image](https://user-images.githubusercontent.com/44462260/123967359-e7bf0000-d9be-11eb-8aec-adc07f9864a9.png)

Reply:
So `integerPart && integerPart.length !== String(parseInt(integerPart)).length` would equal to empty string, `""` and therefore the whole check would be coerced as a false value

Reply:
You're right! Adding `integerPart` as a condition, makes sense. It works!